subgraphs/aave-v2/src/mappings/utilFunctions.ts

```diff
@@ -0,0 +1,559 @@
+  // Get the asset price in Wei and convert it to Eth
+  let assetPriceInUSDC: BigDecimal = BIGDECIMAL_ZERO;
+  const tryAssetPriceInEth = oracle.try_getAssetPrice(tokenAddress);
+  // Fetch USDC price in Wei and convert it to Eth
+  const tryPriceUSDCInEth = oracle.try_getAssetPrice(Address.fromString(USDC_TOKEN_ADDRESS));
+  if (!tryAssetPriceInEth.reverted && !tryPriceUSDCInEth.reverted) {
+    const assetPriceInEth = tryAssetPriceInEth.value;
+    const priceUSDCInEth = tryPriceUSDCInEth.value;
+    assetPriceInUSDC = (new BigDecimal(assetPriceInEth).div(new BigDecimal(priceUSDCInEth)));
+  } else {
+    log.info('REVERTED GETTING USDC PRICE: ASSET? ' + tokenAddress.toHexString(), []);
+  }
+  // Asset price in Eth/USDC priced in Eth = Asset price in in USDC
+  // return price per asset in USDC
+  return assetPriceInUSDC.truncate(3);
+}
+
+export function amountInUSD(priceInUSDC: BigDecimal, decimals: number, amount: BigInt, market: Market): BigDecimal {
+  // This function takes in a token and the amount of the token and converts the amount of that token into USD
+  // Also sets the market input/output token prices to the updated amount
+  const amountInDecimals = bigIntToBigDecimal(amount, <i32>decimals);
+  const amountUSD = amountInDecimals.times(priceInUSDC);
+  log.info(market.id + ' TOKEN AMOUNT ' + amount.toString() + ' TIMES PRICE IN USD ' + priceInUSDC.toString() + ' === ' + amountUSD.toString() + ' ' + market.inputTokens.length.toString(), []);
+  return amountUSD.truncate(3);
+}
+
+export function getCurrentRewardEmissionsUSD(market: Market): BigDecimal[] {
+  // Taking the reward emissions denominated in the reward token, convert it to the value in USD
+  const rewardEmissionsUSD = market.rewardTokenEmissionsUSD;
+  const rewardTokenAddr = Address.fromString(REWARD_TOKEN_ADDRESS);
+  // The DEPOSIT reward token is used as the default. Both the deposit and borrow reward token decimals are the same
+  const rewardToken = loadRewardToken(rewardTokenAddr, RewardTokenType.DEPOSIT);
+  // In the reward emissions arrays index 0 is for the deposit reward, index 1 for the borrow reward
+  const rewardPriceInUSDC = getAssetPriceInUSDC(rewardTokenAddr);
+  rewardEmissionsUSD[0] = amountInUSD(rewardPriceInUSDC, rewardToken.decimals, market.rewardTokenEmissionsAmount[0], market);
+  rewardEmissionsUSD[1] = amountInUSD(rewardPriceInUSDC, rewardToken.decimals, market.rewardTokenEmissionsAmount[1], market);
+  return rewardEmissionsUSD;
+}
+
+// CONTEXT FUNCTIONS
+
+export function getLendingPoolFromCtx(): string {
+  // Get the lending pool/market address with context
+  const context = dataSource.context();
+  return context.getString('lendingPool');
+}
+
+export function getProtocolIdFromCtx(): string {
+  // Get the protocol id with context
+  const context = dataSource.context();
+  return context.getString('protocolId');
+}
+
+// MATH FUNCTIONS
+
+export function calculateRevenues(market: Market, token: Token): void {
```

Comment:
Have you had a chance to compare your numbers against `https://v2.aavewatch.com/`?

Reply:
I just attempted to compare it, however because my deployments are so far off of the current block I cannot compare any data available on aavewatch to the data my subgraph is pulling. What I have been doing as Ive been building this subgraph is logging transaction hashes for events, viewing the transaction data on etherscan, and comparing it to the data saved in the mapping for that particular event. I have also checked contract call data pulled in the mapping to data pulled in the "Read as Proxy" tab for contracts on etherscan.

Reply:
I just attempted to compare it, however because my deployments are so far off of the current block I cannot compare any data available on aavewatch to the data my subgraph is pulling. What I have been doing as Ive been building this subgraph is logging transaction hashes for events, viewing the transaction data on etherscan, and comparing it to the data saved in the mapping for that particular event. I have also checked contract call data pulled in the mapping to data pulled in the "Read as Proxy" tab for contracts on etherscan.